### DISCUSSION TITLE
passage de paramètre à PDOPlusPlus

### DISCUSSION THREAD

**Discussion by laurentsc** [rawsrc/PDOPlusPlus#7] [Q&A]
Bonsoir,
ayant pu récupérer un tableau de select de PPP, je souhaite maintenant passer chaque item du tableau à une méthode de ppp : `$info = [];
        foreach($tab_appli_key as $one_appli_key) {
            $ppp = new PDOPlusPlus();
            $sql =
                "SELECT application_name,platform,platform_owner,publisher from `application` where application_key={$ppp($one_appli_key,'int')}";
   
            $info[] = $ppp->select($sql);
        }

        return $info;`
mais ça se passe mal : xdebug par dans  les choux quand je fais l'affectation de $sql et y a ce message d'erreur : 

>  TypeError: Null or scalar value expected or class with __toString() implemented in C:\projets\ticket_rawsrc\vendor\rawsrc\PDOPlusPlus\PDOPlusPlus.php on line 461

Quel est le problème, STP ?

**Comment by rawsrc** [comment]
Euh, t'es cuit... Tu devrais aller te pieuter et t'y remettre demain matin.
Déjà tu n'as pas besoin d'instancier à chaque boucle une nouvelle instance de `PDOPlusPlus`.
```php
$ppp = new PDOPlusPlus();
$in  = $ppp->injectorInByRef();
$sql = "SELECT application_name, platform, platform_owner, publisher FROM `application` WHERE application_key = {$in($one_appli_key, 'int')}";
foreach ($tab_appli_key as $one_appli_key) {
    $info[] = $ppp->select($sql);
}
return $info;
```
Tu dois lire les exemples, ce cas est largement traité.

**Reply by laurentsc** [reply]
C'est sûr que demain matin, ça ira mieux mais le code que tu me proposes...


> TypeError: Null or scalar value expected or class with __toString() implemented in C:\projets\ticket_rawsrc\vendor\rawsrc\PDOPlusPlus\PDOPlusPlus.php on line 461

Bonne nuit !

**Comment by rawsrc** [comment]
Euh, je ne compte pas déboguer ton code ligne par ligne.
T'as vérifié ce que contenait ta variable : `$tab_appli_key` ?
Je dis ça, je dis rien...

**Comment by laurentsc** [comment]
Bonjour,
oui ce matin, ça allait un peu mieux ; je n'ai plus ce problème par contre, j'ai un autre bug que je n'ai pas encore réussi à corriger...
